.github/workflows/ci.yaml

```diff
@@ -33,13 +50,112 @@ jobs:
+    - name: Push bundle image to ghcr.io for test
+      id: push-bundle-to-ghcr
+      uses: redhat-actions/push-to-registry@v2
+      with:
+        image: cryostat-operator-bundle
+        tags: ${{ steps.build-bundle.outputs.tag }}
+        registry: ghcr.io/${{ github.repository_owner }}
+        username: ${{ github.repository_owner }}
+        password: ${{ secrets.GHCR_PR_TOKEN }}
+    - name: Set up Kind cluster
+      run: |
+        {
+            printf -- "kind: Cluster\n"
+            printf -- "apiVersion: kind.x-k8s.io/v1alpha4\n"
+            printf -- "nodes:\n"
+            printf -- "- role: control-plane\n"
+            printf -- "  kubeadmConfigPatches:\n"
+            printf -- "  - |\n"
+            printf -- "    kind: InitConfiguration\n"
+            printf -- "    nodeRegistration:\n"
+            printf -- "      kubeletExtraArgs:\n"
+            printf -- "        node-labels: \"ingress-ready=true\"\n"
+        } > kind-config.yaml
+        kind create cluster --config=kind-config.yaml -n ci-${{ github.run_id }}
+        # Enabling Ingress
+        kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/main/deploy/static/provider/kind/deploy.yaml
+        kubectl rollout status -w deployment/ingress-nginx-controller -n ingress-nginx --timeout 5m
+    - name: Install Operator Lifecycle Manager
+      run: curl -sL https://github.com/operator-framework/operator-lifecycle-manager/releases/download/v0.24.0/install.sh | bash -s v0.24.0
+    - name: Install Cert Manager
+      run: make cert_manager
+    - uses: docker/login-action@v2
+      with:
+        registry: ${{ env.GHCR_SERVER }}
+        username: ${{ github.repository_owner }}
+        password: ${{ secrets.GHCR_PR_TOKEN }}
+    - name: Set up scorecard environment
+      run: |
+        KUSTOMIZE=$PWD/bin/kustomize
+        make kustomize
+
+        # Create a test namespace
+        kubectl create namespace $SCORECARD_NAMESPACE
+
+        # Generate rbacs
+        pushd internal/images/custom-scorecard-tests/rbac/ && $KUSTOMIZE edit set namespace $SCORECARD_NAMESPACE
+        popd && $KUSTOMIZE build internal/images/custom-scorecard-tests/rbac/ | kubectl apply -f -
+
+        # Define credentials to pull from ghcr.io
+        kubectl create -n $SCORECARD_NAMESPACE secret docker-registry ghcr-key --docker-server="$GHCR_SERVER" \
+          --docker-username="$GHCR_USERNAME" --docker-password="$GHCR_PASSWORD"
+        kubectl patch sa $SCORECARD_SA_NAME -n $SCORECARD_NAMESPACE -p '{"imagePullSecrets": [{"name": "ghcr-key"}]}'
+
+        # Deploy the operator
+        operator-sdk run bundle -n $SCORECARD_NAMESPACE \
+          --timeout 20m --pull-secret-name ghcr-key --service-account $SCORECARD_SA_NAME ${{ steps.push-bundle-to-ghcr.outputs.registry-path }}
+    - name: Run scorecard tests
+      run: |
+        operator-sdk scorecard -n $SCORECARD_NAMESPACE -s $SCORECARD_SA_NAME -w 20m --pod-security=restricted ./bundle
```

Comment:
I wonder if we modify the `test-scorecard` target to support this and reduce duplication. The differences are optionally setting up the pull secret and then using it when deploying the bundle right?

Reply:
I got some difficulty figuring out the clean way for this but I went with: https://github.com/cryostatio/cryostat-operator/pull/588/commits/c0ea55c4ece70b9af5d5202d0189889581054c31.

What do u think?

Reply:
Nice! This looks pretty much what I had in mind. I have a few more suggestions for it.